File: frontend/src/types/diet.ts
export interface Meal {
  id: number;
  meal_type: string;
  title: string;
  calories: number;
  protein: number;
  carbohydrates: number;
  fat: number;
  recipe: string;
  meal_order: number;
}

export interface GroceryItem {
  id: number;
  item_name: string;
  quantity: string;
}

export interface DietPlan {
  id: number;
  title: string;
  goal: string;
  generated_by: string;
  created_at: string;
  meals: Meal[];
  grocery_items: GroceryItem[];
}

export interface GenerateDietRequest {
  days: number;
  regenerate: boolean;
}

export interface ApiResponse<T> {
  message: string;
  data: T;
}

File: backend/app/schemas/diet.py
from datetime import datetime

from pydantic import BaseModel
from pydantic import ConfigDict


# -------------------------------------------------
# Meal
# -------------------------------------------------

class MealResponse(BaseModel):

    model_config = ConfigDict(
        from_attributes=True
    )

    id: int

    meal_type: str

    title: str

    calories: float

    protein: float

    carbohydrates: float

    fat: float

    recipe: str

    meal_order: int


# -------------------------------------------------
# Grocery
# -------------------------------------------------

class GroceryItemResponse(BaseModel):

    model_config = ConfigDict(
        from_attributes=True
    )

    id: int

    item_name: str

    quantity: str


# -------------------------------------------------
# Diet Plan
# -------------------------------------------------

class DietPlanResponse(BaseModel):

    model_config = ConfigDict(
        from_attributes=True
    )

    id: int

    title: str

    goal: str

    generated_by: str

    created_at: datetime

    meals: list[MealResponse]

    grocery_items: list[GroceryItemResponse]

# -------------------------------------------------
# Generate Diet Request
# -------------------------------------------------

class GenerateDietRequest(BaseModel):

    days: int = 1

    regenerate: bool = False

# -------------------------------------------------
# AI Response
# -------------------------------------------------

class GeneratedMeal(BaseModel):

    meal_type: str

    title: str

    calories: float

    protein: float

    carbohydrates: float

    fat: float

    recipe: str


class GeneratedDietResponse(BaseModel):

    meals: list[GeneratedMeal]

    grocery_items: list[str]